rewriter/src/main/java/com/facebook/coresql/rewriter/Rewriter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.rewriter;
+
+import com.facebook.coresql.parser.AstNode;
+import com.facebook.coresql.parser.Unparser;
+import com.facebook.coresql.warning.WarningCode;
+import com.facebook.coresql.warning.WarningCollector;
+
+import static java.util.Objects.requireNonNull;
+
+public abstract class Rewriter
+        extends Unparser
+{
+    private boolean isEnabled;
+    private final WarningCollector collector;
```

Comment:
@kaikalur Do we really want rewrites to be warnings? Do you want users to change the query? If we can handle this automatically users don't really need to do anything right? If there's no action item, why do we need them to be warnings?

Reply:
The idea is we collect them for now. It's upto the caller whether they want to log it or not.

Reply:
Ideally these should be informational in case there are any issues.

Reply:
Leaving unchanged.

Done